### PR TITLE
pref(deps): remove lodash and implement lodash functions used in plugin

### LINKED ISSUE #8: Lodash is the only dependency for this lib!?
I see that you are using lodash and it seems it impacts bundle size:
https://bundlephobia.com/package/@savvywombat/tailwindcss-grid-areas@3.1.0

### PR DESCRIPTION
**Pull request by Hossein-Mirazimi**
fix #8 
I removed the Lodash library from the codebase and implemented the `reduce`, `flatMap`, `uniq`, and `get` functions using pure JavaScript. This helped reduce dependencies, optimize the code, and decrease the bundle size for better performance.

### CODE REVIEW

**Comment by Hossein-Mirazimi** [comment]
@horuskol 
@strothj 

please review this pull request

**Comment by horuskol** [comment]
> @horuskol @strothj
> 
> please review this pull request

Please be patient - it is the weekend and your PR came in at 10pm on a Friday.

**Comment by Hossein-Mirazimi** [comment]
> > @horuskol @strothj
> > please review this pull request
> 
> Please be patient - it is the weekend and your PR came in at 10pm on a Friday.

I apologize if I came across as pushy in my previous message. I assumed that you might not be actively contributing to the plugin due to the time gap since your last contribution

File: pnpm-lock.yaml
Comment (horuskol) [inline pnpm-lock.yaml:1]: Can you remove this lock file?

File: src/util/lodash-fns.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+function reduce(collection, iteratee, initialVal) {
```
Comment (horuskol) [inline src/util/lodash-fns.js:1]: A nitpick: I'd prefer `carry` to `iteratee` here, thanks.

**Comment by horuskol** [comment]
Thanks for your interest and time in putting together the PR. Removing the lodash dependency is something that has been on my list for a while, but I never seem to get the time to do it myself.

If you could resolve the comments I've made on the code changes, I'll see about getting this released very soon.

File: pnpm-lock.yaml
Comment (Hossein-Mirazimi) [reply pnpm-lock.yaml:1]: Oh I forgot this

**Comment by Hossein-Mirazimi** [comment]
> Thanks for your interest and time in putting together the PR. Removing the lodash dependency is something that has been on my list for a while, but I never seem to get the time to do it myself.
> 
> If you could resolve the comments I've made on the code changes, I'll see about getting this released very soon.

I appreciate your quick response and feedback. Thank you again for your time and consideration!

**Comment by Hossein-Mirazimi** [comment]
done

**Comment by Hossein-Mirazimi** [comment]
@horuskol 
hi 
please check this MR

**Comment by horuskol** [comment]
Hi,

Just reviewed and about to merge in the changes. Thanks for you time and help improving this package.

**Pull request merged**